### PR TITLE
Forcing preferred changelog format

### PR DESCRIPTION
**Pull request by a8trejo**
## Description
Forcing preferred changelog format

- When a normal PR with a `changeset` file is merged `changeset-release.yml` will run and create a `Changeset version bump` PR, only the job `changeset-pr-version-bump` runs
- This triggers `changeset-release.yml` **again** but now only `changeset-pr-approve-merge` runs but doesn't run the last `Auto approve PR` step, it however updates the `CHANGELOG.md` file with our preferred format and pushes to the `Changeset version bump`, and then **labels** it `changelog-ready`
- The labeling action triggers `changeset-release.yml` again and runs **only** the `Auto approve PR` step
- Upon manual review and merging of the `changeset` PR, `marketplace-publish.yml` is triggered

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Running the python script locally and printing the formatted changes on the `CHANGELOG.md` file

## Checklist:

- [ ] My code follows the patterns of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Reviewers

----

> [!IMPORTANT]
> Automates changelog formatting and approval process using a Python script and GitHub Actions workflow.
> 
>   - **Script Changes**:
>     - `overwrite_changeset_changelog.py` now reformats existing changelog content if no new content is provided.
>     - Removes first two lines of changeset format and ensures version numbers are bracketed.
>   - **Workflow Changes**:
>     - `changeset-release.yml` automates changelog formatting and labeling.
>     - Adds steps to get current and previous versions, update changelog, commit changes, and push updates.
>     - Adds 'changelog-ready' label after formatting, enabling auto-approval and merge.
>   - **Changelog Config**:
>     - Removes `getReleaseSummary` function from `changelog-config.js`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooVetGit%2FRoo-Cline&utm_source=github&utm_medium=referral)<sup> for bbf7ba7995fbd76d95f2f1b9672a9fa1ca240aeb. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: .changeset/changelog-config.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// Doesn't fully work but '.github/scripts/overwrite_changeset_changelog.py' forces the changelog to be formatted in GHA
```
Comment (joemanley201) [inline .changeset/changelog-config.js:1]: Should we just remove it if it doesn't work?

File: .github/scripts/overwrite_changeset_changelog.py [hunk 2]
```diff
@@ -22,39 +26,37 @@
...
-        
-    Example:
-        >>> content = "## 1.2.0\\nOld changes\\n## 1.1.0\\nOld changes"
-        >>> NEW_CONTENT = "New changes"
-        >>> overwrite_changelog_section(content)
-        '## 1.2.0\\nNew changes\\n## 1.1.0\\nOld changes'
-    """
+NEW_CONTENT = os.environ.get("NEW_CONTENT", "")
+
+def overwrite_changelog_section(changelog_text: str, new_content: str):
```
Comment (joemanley201) [inline .github/scripts/overwrite_changeset_changelog.py:31]: Anyway we can write a test for this? And maybe possible to convert this into a TS file?

Mainly wondering if we have any other python dependencies - if not, i think it is worth to make this in TS and write a test for it as well.

File: .changeset/changelog-config.js (hunk 1 above)
Comment (a8trejo) [reply .changeset/changelog-config.js:1]: it half works, so it's already giving me an easier format to overwrite than the standard `changeset` format from roo-node-types

**Review by joemanley201 — COMMENTED** [review]
Also starting to wonder if this is worth the complexity. What is our current workflow now? Do we manually edit the file?

File: .github/scripts/overwrite_changeset_changelog.py (hunk 2 above)
Comment (a8trejo) [reply .github/scripts/overwrite_changeset_changelog.py:31]: won't need python dependencies, this runs in Github Actions, I find it much easier than a TS file

**Comment by a8trejo** [comment]
> Also starting to wonder if this is worth the complexity. What is our current workflow now? Do we manually edit the file?

I think it is, we'll need less manual edits if we just click merge

**Comment by a8trejo** [comment]
> > Also starting to wonder if this is worth the complexity. What is our current workflow now? Do we manually edit the file?
> 
> I think it is, we'll need less manual edits if we just click merge

also, is ready to merge and wasn't that complex

**Comment by a8trejo** [comment]
> > > Also starting to wonder if this is worth the complexity. What is our current workflow now? Do we manually edit the file?
> > 
> > 
> > I think it is, we'll need less manual edits if we just click merge
> 
> also, is ready to merge and wasn't that complex

the current workfow is manually running git pull & checkout into the changeset branch, updating the CHANGELOG file with the proper format, running git add, git commit, git push, and then clicking merge, 

I think is worth it to reduce all of that to just clicking merge

File: .changeset/changelog-config.js (hunk 1 above)
Comment (joemanley201) [reply .changeset/changelog-config.js:1]: Can we remove the `getReleaseSummary` function then? That doesn't work right?

**Review by joemanley201 — APPROVED** [review]
Left 1 comment about removing the things that are not working in `.changeset/changelog-config.js` - mainly to make it super obvious about what is working and what is not working.

Looks good otherwise 🙌

File: .changeset/changelog-config.js (hunk 1 above)
Comment (a8trejo) [reply .changeset/changelog-config.js]: removed 👍

**Pull request merged**